stor/obs.py

```diff
@@ -450,6 +452,22 @@ def close(self):
                 self.flush()
             self._buffer.close()
         self.closed = True
+        # we want to wait_on_close on DXPath only if no error was thrown while writing the file
+        if not any(sys.exc_info()):  # pragma: no cover
+            self.dx_close_helper()
+
+    def dx_close_helper(self):
+        if isinstance(self._path, stor.dx.DXPath):
+            wait_on_close = stor.settings.get()['dx']['wait_on_close']
+            if wait_on_close:
+                f = dxpy.DXFile(dxid=self._path.canonical_resource,
+                                project=self._path.canonical_project)
+                try:
+                    f.wait_on_close(wait_on_close)
+                # ignore timeout because we want client code to deal with it
+                except dxpy.DXError as e:  # pragma: no cover
```

Comment:
nit: `dxpy.DXFileError` according to docs

Reply:
Manually made a file reach timeout error, as well as checked the code. It raises DXError, so probably inconsistency in doc.

Search for `_wait_on_close` here: http://autodoc.dnanexus.com/bindings/python/current/_modules/dxpy/bindings.html